src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -139,14 +148,66 @@ private static Resource parseResource(String resourceStr) {
 
         Matcher matcher = SERVICE_RESOURCE_PATTERN.matcher(resourceStr);
         if (matcher.matches()) {
-            return Resource.builder().service(matcher.group(1)).resourceType(matcher.group(2))
-                    .resourceName(matcher.group(3)).build();
+            return Resource.builder().service(matcher.group(1))
+                    .resourceType(matcher.group(2))
+                    .resourceName(matcher.group(3))
+                    .build();
         }
 
         throw new IllegalArgumentException(
                 String.format("Resource %s is not in the form of %s", resourceStr, SERVICE_RESOURCE_PATTERN.pattern()));
     }
 
+    /**
+     * utility method to permission with variable value.
+     *
+     * @param session    current device session
+     * @param permission permission to parse
+     * @return updated permission
+     */
+
+    static Permission updateResource(@NonNull Session session, @NonNull Permission permission) {
+
+        String resource = permission.getResource();
+
+        Matcher matcher = POLICY_VARIABLE_PATTERN.matcher(resource);
+
+        while (matcher.find()) {
+
+            String policyVariable = matcher.group(1);
+            String[] vars = policyVariable.split("\\.");
+            String attributeNamespace = vars[1];
+            String attributeName = vars[2];
+
+            // this currently supports the ThingName attribute
+            if (policyVariable.equals(THING_NAME_VARIABLE)
```

Comment:
flip this equals around to `THING_NAME_VARIABLE.equals(policyVariable)` that way if policy variable is somehow null, this wouldn't error.